plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts

```diff
@@ -1,31 +1,35 @@
 /**
  * External dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { STORE_KEY } from '~/marketing/data/constants';
 import { Plugin } from './types';
 
+const selector = 'getRecommendedPlugins';
 const category = 'marketing';
 
-type SelectResult = {
-	isLoading: boolean;
-	plugins: Plugin[];
-};
-
 export const useRecommendedPlugins = () => {
-	return useSelect< SelectResult >(
-		( select ) => {
-			const { getRecommendedPlugins, isResolving } = select( STORE_KEY );
+	const { invalidateResolution } = useDispatch( STORE_KEY );
+
+	const refetch = () => {
+		invalidateResolution( selector, [ category ] );
+	};
+
+	return useSelect( ( select ) => {
+		const { getRecommendedPlugins, hasFinishedResolution } =
+			select( STORE_KEY );
+		const plugins = getRecommendedPlugins< Plugin[] >( category );
 
-			return {
-				isLoading: isResolving( 'getRecommendedPlugins', [ category ] ),
-				plugins: getRecommendedPlugins( category ),
-			};
-		},
-		[ category ]
-	);
+		return {
+			isLoading:
```

Comment:
After install there's a very brief moment where the activated plugin remains in the list and buttons are no longer disabled.

Optionally, I think that creating separate `isLoading` and `isInitializing` states may help differentiate and disable the UI while being refetched.  Something like:

```
isInitializing: ! hasFinishedResolution( selector, [ category ] ) &&
				! plugins.length,
isLoading: ! hasFinishedResolution( selector, [ category ] );
```